src/libcore/iter/range.rs

```diff
@@ -353,6 +304,23 @@ impl<A: Step> Iterator for ops::RangeFrom<A> where
     fn size_hint(&self) -> (usize, Option<usize>) {
         (usize::MAX, None)
     }
+
+    #[inline]
+    fn count(self) -> usize {
+        usize::MAX
```

Comment:
@ollie27 I’m fairly confident that `nth` maintains the same behaviour as the current `next` by being ever-so-slightly broken around the last value.